src/main/com/fulcrologic/fulcro/components.cljc

```diff
@@ -1091,6 +1091,7 @@
                      (string? class-or-factory) class-or-factory
                      (some-> class-or-factory meta (contains? :queryid)) (some-> class-or-factory meta :queryid)
                      :otherwise (query-id class-or-factory nil))]
+    (util/check-query-valid (get-query class-or-factory) component-name)
```

Comment:
you have to pass the state map to get query to actually see the new dynamic query

`(get-query c-of-r @state-atom)`